Reqnroll.VisualStudio/Editor/Commands/FindUnusedStepDefinitionsCommand.cs

```diff
@@ -0,0 +1,206 @@
+            Finished.Set();
+        }
+        private async Task FindUsagesInternalAsync(IProjectScope[] reqnrollTestProjects, string fileName,
+    SnapshotPoint triggerPoint, IAsyncContextMenu asyncContextMenu, CancellationToken cancellationToken,
+    UnusedStepDefinitionSummary summary)
+        {
+            foreach (var project in reqnrollTestProjects)
+            {
+                if (cancellationToken.IsCancellationRequested)
+                    break;
+
+                var stepDefinitions = await GetStepDefinitionsAsync(project, fileName, triggerPoint);
+                //summary.FoundStepDefinitions += stepDefinitions.Length;
+                if (stepDefinitions.Length == 0)
+                    continue;
+
+                var featureFiles = project.GetProjectFiles(".feature");
+                var configuration = project.GetDeveroomConfiguration();
+                var projectUnusedStepDefinitions = _stepDefinitionsUnusedFinder.FindUnused(stepDefinitions, featureFiles, configuration);
+                foreach (var unused in projectUnusedStepDefinitions)
+                {
+                    if (cancellationToken.IsCancellationRequested)
+                        break;
+
+                    //await Task.Delay(500);
+
+                    //TODO: create menu item - what type of menu item?
+                    asyncContextMenu.AddItems(CreateMenuItem(unused, project));
+                    summary.UnusedStepDefinitions++;
+                }
+
+                summary.ScannedFeatureFiles += featureFiles.Length;
+            }
+        }
+
+        private async Task<ProjectStepDefinitionBinding[]> GetStepDefinitionsAsync(IProjectScope project, string fileName,
+     SnapshotPoint triggerPoint)
+        {
+            var discoveryService = project.GetDiscoveryService();
+            var bindingRegistry = await discoveryService.BindingRegistryCache.GetLatest();
+            if (bindingRegistry == ProjectBindingRegistry.Invalid)
+                Logger.LogWarning(
+                    $"Unable to get step definitions from project '{project.ProjectName}', usages will not be found for this project.");
+            return GetStepDefinitions(fileName, triggerPoint, bindingRegistry);
+        }
+
+        //TODO: since this Command doesn't need to filter the set of StepDefinitions, this can be merged with the method above
+        internal static ProjectStepDefinitionBinding[] GetStepDefinitions(string fileName, SnapshotPoint triggerPoint,
+            ProjectBindingRegistry bindingRegistry)
+        {
+            return bindingRegistry.StepDefinitions
+                    //                .Where(sd => sd.Implementation?.SourceLocation != null &&
+                    //                             sd.Implementation.SourceLocation.SourceFile == fileName &&
+                    //
+                    .ToArray();
+        }
+
+        //TODO: near duplicate of similar method in FindStepDefinitionUsagesCommand
```

Comment:
I agree. Maybe try to create a `JumpListEditorCommand` abstract base class for this two commands?

Reply:
After looking at it further, decided against it for now. The other is dealing with creating a menu item and Jump for feature files, not C# files. I found that the base class for these Editor Commands already has an implementation for PerformJump for C# code. So I refactored the FindUnusedStepDefinitionsCommand to use the base class behavior. The implementation that I was previously looking at, in FindStepDefinitionUsages, shadows the base class implementation because jumping to a feature file location is somewhat different.

Let me know if you'd like me to further refactor this such that both classes use a common implementation.

Reply:
Makes sense. All good.